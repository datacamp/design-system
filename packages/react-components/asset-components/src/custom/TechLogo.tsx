import React from 'react';

import {
  Git,
  Python,
  R,
  Scala,
  Shell,
  Spreadsheet,
  Sql,
  Tableau,
  Theory,
} from '../generated/technologies';

export const logoComponents = {
  Git,
  python: Python,
  R,
  Scala,
  Shell,
  Spreadsheets: Spreadsheet,
  SQL: Sql,
  Tableau,
  Theory,
};

export type Technology = keyof typeof logoComponents;

interface TechLogoProps extends React.ComponentProps<typeof Git> {
  technology: Technology;
}

const TechLogo: React.FC<TechLogoProps> = ({ technology, ...otherProps }) => {
  const Logo = logoComponents[technology] || Python;
  return <Logo {...otherProps} />;
};

export default TechLogo;
