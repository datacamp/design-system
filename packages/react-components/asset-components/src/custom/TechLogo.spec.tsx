import axeRender from '@datacamp/waffles-axe-render';

import TechLogo, { logoComponents, Technology } from './TechLogo';

describe('<TechLogo />', () => {
  (Object.keys(logoComponents) as Technology[]).forEach(technology => {
    it(`renders a ${technology} logo`, async () => {
      const { container } = await axeRender(
        <TechLogo technology={technology} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
