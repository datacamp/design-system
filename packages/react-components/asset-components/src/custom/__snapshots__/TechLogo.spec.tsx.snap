// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TechLogo /> renders a Excel logo 1`] = `
<svg
  viewBox="0 0 39 39"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <circle
      cx="19.5"
      cy="19.5"
      fill="#05192D"
      r="19.5"
    />
    <path
      d="M20.971 9.016l-.001 1.986h8.28a.75.75 0 01.743.652l.007.102v14.502a.751.751 0 01-.75.753h-8.285v2.005L9 26.842V11.158l11.971-2.142zm-.001 4.06h2.832v1.817h-2.833v.693h2.833v1.817l-2.834-.001v.693h2.834v1.817h-2.835v.692h2.835v1.818l-2.836-.001v.697h2.836v1.734l-2.837-.001v1.468h8.343V11.691H20.97v1.385zm6.875 10.042v1.73H24.49v-1.73h3.355zM17.143 15.47l-1.476.068-1.061 2.506-.882-2.412-1.516.072 1.563 3.31-1.646 3.274 1.402.08 1.025-2.492 1.047 2.61 1.591.094-1.76-3.566 1.713-3.544zm10.702 5.135v1.817H24.49v-1.817h3.355zm0-2.51v1.817H24.49v-1.817h3.355zm0-2.509v1.817H24.49v-1.817h3.355zm0-2.51v1.817H24.49v-1.816h3.355z"
      fill="#FFF"
      fill-rule="nonzero"
    />
  </g>
</svg>
`;

exports[`<TechLogo /> renders a Git logo 1`] = `
<svg
  viewBox="0 0 39 39"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <circle
      cx="19.5"
      cy="19.5"
      fill="#05192D"
      r="19.5"
    />
    <path
      d="M22.11 8.538l8.643 8.643a3.279 3.279 0 010 4.638l-8.643 8.643a3.279 3.279 0 01-4.637 0l-8.644-8.643a3.279 3.279 0 010-4.638l6.086-6.086 2.997 3.097a2.002 2.002 0 001.109 2.537v6.157a2.002 2.002 0 101.54 0v-5.954l2.316 2.395a2.001 2.001 0 101.103-1.073l-2.4-2.474a2.001 2.001 0 00-2.633-2.713l-2.955-3.05 1.48-1.479a3.279 3.279 0 014.638 0z"
      fill="#FFF"
    />
  </g>
</svg>
`;

exports[`<TechLogo /> renders a Power BI logo 1`] = `
<svg
  viewBox="0 0 39 39"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <circle
      cx="19.5"
      cy="19.5"
      fill="#05192D"
      r="19.5"
    />
    <path
      d="M14.874 24.518c.05.521.021 1.051.024 1.578.002.494.013.99-.005 1.484-.03.861-.588 1.437-1.354 1.42-.755-.017-1.236-.58-1.241-1.458-.003-.462 0-.925 0-1.388 0-.495-.006-.99.001-1.484.01-.758.506-1.329 1.19-1.381.692-.053 1.313.47 1.385 1.229zm2.648-5.973c.783-.007 1.315.618 1.317 1.556l.002 1.844v1.844c0 1.23.002 2.46-.002 3.689-.002.908-.525 1.515-1.299 1.521-.783.006-1.306-.586-1.307-1.503-.005-2.475-.004-4.95 0-7.424 0-.922.513-1.52 1.29-1.527zm3.95 2c.762.002 1.291.592 1.3 1.473.008.942.001 1.884.001 2.827 0 .894.007 1.788-.001 2.683-.009.88-.537 1.47-1.3 1.47-.755.002-1.296-.598-1.3-1.47a669.57 669.57 0 01.001-5.51c.004-.877.544-1.474 1.3-1.473zm3.93-5.266c.784-.004 1.3.585 1.302 1.509.004 1.787 0 3.574 0 5.361 0 1.82.006 3.639-.004 5.458-.004.68-.394 1.181-.983 1.325a1.24 1.24 0 01-1.452-.71 1.89 1.89 0 01-.148-.737c-.01-3.575-.01-7.15-.004-10.724.002-.88.538-1.478 1.29-1.482zm-8.627-4.276h11.288c1.792.002 2.936 1.208 2.936 3.091 0 3.195.002 6.39 0 9.585-.002 1.63-1.222 2.91-2.781 2.938-.312.006-.5-.064-.469-.447.025-.308-.077-.63.426-.643 1.155-.027 1.794-.737 1.796-1.942.005-3.195.006-6.39 0-9.585-.003-1.206-.705-1.938-1.859-1.938-5.743-.004-11.485-.005-17.228 0-1.142 0-1.835.72-1.838 1.9-.005 3.243-.006 6.486.001 9.729.003 1.045.652 1.797 1.611 1.835.647.025.606-.02.604.726 0 .093-.019.186-.028.273-1.151.348-2.585-.477-3.045-1.769a3.164 3.164 0 01-.18-1.025c-.013-3.274-.009-6.55-.007-9.824.001-1.633 1.179-2.89 2.741-2.9 2.01-.013 4.022-.004 6.032-.004z"
      fill="#FFF"
    />
  </g>
</svg>
`;

exports[`<TechLogo /> renders a PowerBi logo 1`] = `
<svg
  viewBox="0 0 39 39"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <circle
      cx="19.5"
      cy="19.5"
      fill="#05192D"
      r="19.5"
    />
    <path
      d="M14.874 24.518c.05.521.021 1.051.024 1.578.002.494.013.99-.005 1.484-.03.861-.588 1.437-1.354 1.42-.755-.017-1.236-.58-1.241-1.458-.003-.462 0-.925 0-1.388 0-.495-.006-.99.001-1.484.01-.758.506-1.329 1.19-1.381.692-.053 1.313.47 1.385 1.229zm2.648-5.973c.783-.007 1.315.618 1.317 1.556l.002 1.844v1.844c0 1.23.002 2.46-.002 3.689-.002.908-.525 1.515-1.299 1.521-.783.006-1.306-.586-1.307-1.503-.005-2.475-.004-4.95 0-7.424 0-.922.513-1.52 1.29-1.527zm3.95 2c.762.002 1.291.592 1.3 1.473.008.942.001 1.884.001 2.827 0 .894.007 1.788-.001 2.683-.009.88-.537 1.47-1.3 1.47-.755.002-1.296-.598-1.3-1.47a669.57 669.57 0 01.001-5.51c.004-.877.544-1.474 1.3-1.473zm3.93-5.266c.784-.004 1.3.585 1.302 1.509.004 1.787 0 3.574 0 5.361 0 1.82.006 3.639-.004 5.458-.004.68-.394 1.181-.983 1.325a1.24 1.24 0 01-1.452-.71 1.89 1.89 0 01-.148-.737c-.01-3.575-.01-7.15-.004-10.724.002-.88.538-1.478 1.29-1.482zm-8.627-4.276h11.288c1.792.002 2.936 1.208 2.936 3.091 0 3.195.002 6.39 0 9.585-.002 1.63-1.222 2.91-2.781 2.938-.312.006-.5-.064-.469-.447.025-.308-.077-.63.426-.643 1.155-.027 1.794-.737 1.796-1.942.005-3.195.006-6.39 0-9.585-.003-1.206-.705-1.938-1.859-1.938-5.743-.004-11.485-.005-17.228 0-1.142 0-1.835.72-1.838 1.9-.005 3.243-.006 6.486.001 9.729.003 1.045.652 1.797 1.611 1.835.647.025.606-.02.604.726 0 .093-.019.186-.028.273-1.151.348-2.585-.477-3.045-1.769a3.164 3.164 0 01-.18-1.025c-.013-3.274-.009-6.55-.007-9.824.001-1.633 1.179-2.89 2.741-2.9 2.01-.013 4.022-.004 6.032-.004z"
      fill="#FFF"
    />
  </g>
</svg>
`;

exports[`<TechLogo /> renders a Python logo 1`] = `
<svg
  viewBox="0 0 39 39"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <circle
      cx="19.5"
      cy="19.5"
      fill="#05192D"
      r="19.5"
    />
    <path
      d="M26.849 14.552c1.432 0 2.107 1.055 2.46 2.457.492 1.948.514 3.408 0 4.926-.497 1.474-1.03 2.457-2.46 2.457h-7.376v.615h4.915v1.847c0 1.399-1.221 2.11-2.46 2.463-1.864.533-3.358.451-4.915 0-1.301-.377-2.46-1.15-2.46-2.463V22.24c0-1.328 1.113-2.463 2.46-2.463h4.915c1.638 0 3.074-1.404 3.074-3.073v-2.152zm-5.166 11.351a1.018 1.018 0 101.018 1.015c0-.561-.455-1.015-1.018-1.015zm-2.27-16.59a15.51 15.51 0 012.57.2c1.337.22 2.465 1.215 2.465 2.538v4.65a2.472 2.472 0 01-2.466 2.482h-4.925c-1.672 0-3.08 1.422-3.08 3.033v2.232H12.28c-1.433 0-2.27-1.03-2.62-2.476-.473-1.942-.453-3.103 0-4.963.392-1.623 1.647-2.476 3.08-2.476h6.782v-.62h-4.931V12.05c0-1.409.378-2.173 2.465-2.538a14.217 14.217 0 012.356-.2zm-2.678 1.747a1.02 1.02 0 00-1.019 1.021c0 .562.456 1.016 1.02 1.016a1.018 1.018 0 100-2.037z"
      fill="#FFF"
      fill-rule="nonzero"
    />
  </g>
</svg>
`;

exports[`<TechLogo /> renders a R logo 1`] = `
<svg
  viewBox="0 0 39 39"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <circle
      cx="19.5"
      cy="19.5"
      fill="#05192D"
      r="19.5"
    />
    <g
      fill="#FFF"
      fill-rule="nonzero"
    >
      <path
        d="M19.47 25.661c-6.87 0-12.439-3.73-12.439-8.33C7.031 12.73 12.601 9 19.471 9c6.869 0 12.438 3.73 12.438 8.33 0 4.602-5.57 8.331-12.439 8.331zm1.904-13.404c-5.222 0-9.455 2.55-9.455 5.695 0 3.146 4.233 5.695 9.455 5.695s9.075-1.743 9.075-5.695c0-3.95-3.853-5.695-9.075-5.695z"
      />
      <path
        d="M25.975 21.903s.753.227 1.19.448c.152.077.415.23.604.432.186.197.276.396.276.396l2.967 5.003-4.796.002-2.242-4.211s-.46-.789-.742-1.018c-.235-.19-.336-.258-.569-.258h-1.14l.002 5.484-4.244.002v-14.01h8.522s3.88.07 3.88 3.763c0 3.692-3.708 3.967-3.708 3.967zm-1.846-4.691l-2.569-.002-.001 2.382h2.57s1.19-.004 1.19-1.212c0-1.233-1.19-1.168-1.19-1.168z"
      />
    </g>
  </g>
</svg>
`;

exports[`<TechLogo /> renders a SQL logo 1`] = `
<svg
  viewBox="0 0 39 39"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <circle
      cx="19.5"
      cy="19.5"
      fill="#05192D"
      r="19.5"
    />
    <path
      d="M13.132 15.92a.5.5 0 01-.271.072.65.65 0 01-.341-.117c-.112-.07-.242-.147-.39-.23a3.127 3.127 0 00-.515-.227 2.123 2.123 0 00-.69-.1c-.24 0-.447.029-.623.087-.174.057-.32.137-.438.237a.93.93 0 00-.261.358c-.058.14-.087.296-.087.465a.79.79 0 00.171.521c.119.145.278.268.477.372.205.106.438.2.7.285.259.083.53.174.816.274.28.098.555.211.823.34.273.132.516.298.731.5.216.203.39.451.52.744.132.295.197.65.197 1.065 0 .45-.076.873-.228 1.267a2.95 2.95 0 01-.67 1.034 3.15 3.15 0 01-1.074.692c-.422.168-.904.252-1.445.252a4.755 4.755 0 01-1.788-.357 4.505 4.505 0 01-.779-.412 3.855 3.855 0 01-.648-.537l-.043-.045.032-.053.524-.878c.054-.07.12-.13.197-.178a.504.504 0 01.271-.076c.129 0 .263.052.405.153.13.093.28.195.452.305.17.11.372.21.604.3.227.088.502.133.826.133.492 0 .868-.117 1.132-.348.26-.23.391-.559.391-.995 0-.24-.058-.431-.172-.578a1.455 1.455 0 00-.477-.387 3.643 3.643 0 00-.696-.273c-.267-.077-.538-.16-.814-.25a6.993 6.993 0 01-.82-.328 2.7 2.7 0 01-.733-.508 2.326 2.326 0 01-.52-.785c-.131-.314-.196-.7-.196-1.155a2.778 2.778 0 01.846-1.994 3.092 3.092 0 011.012-.644c.4-.161.856-.242 1.366-.242.576 0 1.109.092 1.597.275.488.183.905.441 1.25.773l.043.041-.466.91a.578.578 0 01-.198.241zm7.428 7.843a5.73 5.73 0 01-.751.048 5.219 5.219 0 01-2.022-.38 4.593 4.593 0 01-1.569-1.061 4.768 4.768 0 01-1.011-1.61 5.558 5.558 0 01-.358-2.015c0-.725.12-1.397.358-2.016a4.77 4.77 0 011.011-1.609 4.593 4.593 0 011.57-1.06 5.219 5.219 0 012.02-.38c.738 0 1.41.127 2.019.383.608.256 1.13.61 1.562 1.06.433.451.77.987 1.008 1.606.239.619.358 1.291.358 2.016 0 .442-.045.865-.135 1.269-.09.404-.22.782-.389 1.134a4.748 4.748 0 01-.618.97 4.442 4.442 0 01-.76.727l2.502 2.767h-1.657c-.22 0-.421-.03-.6-.092a1.204 1.204 0 01-.503-.346l-1.376-1.542a5.187 5.187 0 01-.659.13zm2.2-5.019c0-.527-.068-.998-.205-1.414a2.979 2.979 0 00-.586-1.055 2.516 2.516 0 00-.925-.66 3.163 3.163 0 00-1.236-.23c-.46 0-.873.077-1.24.23a2.539 2.539 0 00-.931.66 2.99 2.99 0 00-.592 1.055 4.449 4.449 0 00-.209 1.414c0 .528.07.999.209 1.415.138.414.335.764.591 1.05.256.286.567.505.933.658.366.153.78.23 1.239.23.46 0 .872-.077 1.236-.23.364-.153.672-.372.925-.658a2.94 2.94 0 00.586-1.05c.137-.417.205-.888.205-1.415zm9.255 3.3v1.659h-5.797v-9.916h1.948v8.257h3.849z"
      fill="#FFF"
      fill-rule="nonzero"
    />
  </g>
</svg>
`;

exports[`<TechLogo /> renders a Scala logo 1`] = `
<svg
  viewBox="0 0 39 39"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <circle
      cx="19.5"
      cy="19.5"
      fill="#05192D"
      r="19.5"
    />
    <path
      d="M26.292 7v5.317s0 2.215-13.292 3.545v-5.317S26.292 9.215 26.292 7M13 17.634s13.292-1.33 13.292-3.545v5.317s0 2.216-13.292 3.545v-5.317zm0 12.406v-5.317s13.292-1.33 13.292-3.545v5.317s0 2.216-13.292 3.545"
      fill="#FFF"
      fill-rule="nonzero"
    />
  </g>
</svg>
`;

exports[`<TechLogo /> renders a Shell logo 1`] = `
<svg
  viewBox="0 0 39 39"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <circle
      cx="19.5"
      cy="19.5"
      fill="#05192D"
      r="19.5"
    />
    <path
      d="M30.269 21.537v3.202h-11.06v-3.202h11.06zM9.022 12.806l9.72 4.041v2.534l-9.72 4.05v-3.022l5.908-2.286-5.908-2.278v-3.039z"
      fill="#FFF"
    />
  </g>
</svg>
`;

exports[`<TechLogo /> renders a Spreadsheets logo 1`] = `
<svg
  viewBox="0 0 39 39"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <circle
      cx="19.5"
      cy="19.5"
      fill="#05192D"
      r="19.5"
    />
    <path
      d="M27.964 9.604a2.304 2.304 0 012.304 2.304v15.183a2.304 2.304 0 01-2.304 2.304H11.035a2.304 2.304 0 01-2.304-2.304V11.908a2.304 2.304 0 012.304-2.304h16.93zm-10.076 13.68h-5.962a.576.576 0 00-.576.575v1.881c0 .318.258.576.576.576h5.962a.576.576 0 00.576-.576v-1.88a.576.576 0 00-.576-.577zm9.313 0H21.24a.576.576 0 00-.576.575v1.881c0 .318.258.576.576.576H27.2a.576.576 0 00.576-.576v-1.88a.576.576 0 00-.576-.577zm-9.313-5.24h-5.962a.576.576 0 00-.576.576v1.881c0 .319.258.576.576.576h5.962a.576.576 0 00.576-.576v-1.88a.576.576 0 00-.576-.577zm9.313 0H21.24a.576.576 0 00-.576.576v1.881c0 .319.258.576.576.576H27.2a.576.576 0 00.576-.576v-1.88a.576.576 0 00-.576-.577zm-9.313-5.239h-5.962a.576.576 0 00-.576.576v1.882c0 .318.258.576.576.576h5.962a.576.576 0 00.576-.576V13.38a.576.576 0 00-.576-.576zm9.313 0H21.24a.576.576 0 00-.576.576v1.882c0 .318.258.576.576.576H27.2a.576.576 0 00.576-.576V13.38a.576.576 0 00-.576-.576z"
      fill="#FFF"
    />
  </g>
</svg>
`;

exports[`<TechLogo /> renders a Sql logo 1`] = `
<svg
  viewBox="0 0 39 39"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <circle
      cx="19.5"
      cy="19.5"
      fill="#05192D"
      r="19.5"
    />
    <path
      d="M13.132 15.92a.5.5 0 01-.271.072.65.65 0 01-.341-.117c-.112-.07-.242-.147-.39-.23a3.127 3.127 0 00-.515-.227 2.123 2.123 0 00-.69-.1c-.24 0-.447.029-.623.087-.174.057-.32.137-.438.237a.93.93 0 00-.261.358c-.058.14-.087.296-.087.465a.79.79 0 00.171.521c.119.145.278.268.477.372.205.106.438.2.7.285.259.083.53.174.816.274.28.098.555.211.823.34.273.132.516.298.731.5.216.203.39.451.52.744.132.295.197.65.197 1.065 0 .45-.076.873-.228 1.267a2.95 2.95 0 01-.67 1.034 3.15 3.15 0 01-1.074.692c-.422.168-.904.252-1.445.252a4.755 4.755 0 01-1.788-.357 4.505 4.505 0 01-.779-.412 3.855 3.855 0 01-.648-.537l-.043-.045.032-.053.524-.878c.054-.07.12-.13.197-.178a.504.504 0 01.271-.076c.129 0 .263.052.405.153.13.093.28.195.452.305.17.11.372.21.604.3.227.088.502.133.826.133.492 0 .868-.117 1.132-.348.26-.23.391-.559.391-.995 0-.24-.058-.431-.172-.578a1.455 1.455 0 00-.477-.387 3.643 3.643 0 00-.696-.273c-.267-.077-.538-.16-.814-.25a6.993 6.993 0 01-.82-.328 2.7 2.7 0 01-.733-.508 2.326 2.326 0 01-.52-.785c-.131-.314-.196-.7-.196-1.155a2.778 2.778 0 01.846-1.994 3.092 3.092 0 011.012-.644c.4-.161.856-.242 1.366-.242.576 0 1.109.092 1.597.275.488.183.905.441 1.25.773l.043.041-.466.91a.578.578 0 01-.198.241zm7.428 7.843a5.73 5.73 0 01-.751.048 5.219 5.219 0 01-2.022-.38 4.593 4.593 0 01-1.569-1.061 4.768 4.768 0 01-1.011-1.61 5.558 5.558 0 01-.358-2.015c0-.725.12-1.397.358-2.016a4.77 4.77 0 011.011-1.609 4.593 4.593 0 011.57-1.06 5.219 5.219 0 012.02-.38c.738 0 1.41.127 2.019.383.608.256 1.13.61 1.562 1.06.433.451.77.987 1.008 1.606.239.619.358 1.291.358 2.016 0 .442-.045.865-.135 1.269-.09.404-.22.782-.389 1.134a4.748 4.748 0 01-.618.97 4.442 4.442 0 01-.76.727l2.502 2.767h-1.657c-.22 0-.421-.03-.6-.092a1.204 1.204 0 01-.503-.346l-1.376-1.542a5.187 5.187 0 01-.659.13zm2.2-5.019c0-.527-.068-.998-.205-1.414a2.979 2.979 0 00-.586-1.055 2.516 2.516 0 00-.925-.66 3.163 3.163 0 00-1.236-.23c-.46 0-.873.077-1.24.23a2.539 2.539 0 00-.931.66 2.99 2.99 0 00-.592 1.055 4.449 4.449 0 00-.209 1.414c0 .528.07.999.209 1.415.138.414.335.764.591 1.05.256.286.567.505.933.658.366.153.78.23 1.239.23.46 0 .872-.077 1.236-.23.364-.153.672-.372.925-.658a2.94 2.94 0 00.586-1.05c.137-.417.205-.888.205-1.415zm9.255 3.3v1.659h-5.797v-9.916h1.948v8.257h3.849z"
      fill="#FFF"
      fill-rule="nonzero"
    />
  </g>
</svg>
`;

exports[`<TechLogo /> renders a Tableau logo 1`] = `
<svg
  viewBox="0 0 39 39"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <circle
      cx="19.5"
      cy="19.5"
      fill="#05192D"
      r="19.5"
    />
    <path
      d="M20.03 26.356v2.365h2.089v.913h-2.09V32h-.995v-2.366h-2.089v-.913h2.09v-2.365h.994zm-6.396-4.79v3.391h3.08v1.034h-3.08v3.393H12.47V25.99h-3.08v-1.034h3.08v-3.39h1.165zm12.921 0v3.391h3.08v1.034h-3.08v3.393h-1.167V25.99H22.31v-1.034h3.08v-3.39h1.166zm-6.367-6.581v3.788h3.437v1.346h-3.437v3.787h-1.352V20.12H15.4v-1.346h3.437v-3.788h1.352zm9.723 1.64v2.364H32v.914h-2.09v2.365h-.994v-2.365h-2.09v-.914h2.09v-2.365h.995zm-20.107.18v2.275h2.074v.702l-2.074.013v2.293h-.73v-2.289L7 19.812v-.732h2.075v-2.275h.729zm16.751-7.387v3.408h3.08v1.033h-3.08v3.376h-1.167v-3.376H22.31v-1.033h3.08V9.418h1.166zm-12.957 0v3.407h3.116v1.003h-3.116v3.407h-1.092v-3.407H9.39v-1.003h3.116V9.418h1.092zM19.903 7v2.32h2.067v.703h-2.067v2.32h-.743v-2.32h-2.067v-.702h2.067V7h.743z"
      fill="#FFF"
    />
  </g>
</svg>
`;

exports[`<TechLogo /> renders a Theory logo 1`] = `
<svg
  viewBox="0 0 39 39"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <circle
      cx="19.261"
      cy="19.261"
      fill="#05192D"
      r="19.261"
    />
    <path
      d="M23.791 7.945c1.279 0 2.402.663 3.047 1.663a4.112 4.112 0 00-1.402.703l-.203.166-.022.004a.65.65 0 00-.098.037c-.742.35-1.33.4-1.797.196a.645.645 0 00-.518 1.181c.404.178.835.25 1.29.224a4.18 4.18 0 00-.252 1.06.645.645 0 001.284.133 2.857 2.857 0 012.094-2.47l.175-.041.148.012a3.624 3.624 0 013.212 4.058c-1.796-.537-3.785.22-4.788 1.959a.566.566 0 00-.066.172c-.421-.054-.891-.01-1.401.132a.645.645 0 10.347 1.242c.946-.264 1.443-.064 1.737.638a.645.645 0 001.19-.498 2.75 2.75 0 00-.568-.88l-.142-.131.02-.03c.696-1.207 2.053-1.731 3.284-1.373l-.006.01a3.626 3.626 0 01.037 6.605 3.626 3.626 0 01-2.388 5.151l.039-.008c-1.248.282-2.078-.194-2.682-1.595a.758.758 0 00-.997-.393.757.757 0 00-.4.994c.612 1.416 1.538 2.273 2.715 2.513a3.624 3.624 0 01-6.506-1.995l-.005-.192v-4.585l1.71.001c.125.473.41.91.859 1.297a.645.645 0 10.842-.977c-.612-.528-.65-1.102-.074-1.97a.645.645 0 10-1.075-.714 4.15 4.15 0 00-.446.863l-.07.21H20.17V11.268l.012.007a3.624 3.624 0 013.61-3.33zm-9.061 0a3.624 3.624 0 013.61 3.33l.012-.007v3.419h-.036c-2.154.114-3.443-.631-4.028-2.29l-.064-.197a.645.645 0 00-1.234.38c.71 2.312 2.47 3.466 5.118 3.405l.244-.01v11.217a3.624 3.624 0 01-6.635 2.014 2.882 2.882 0 012.59-4.147.645.645 0 100-1.29c-.476 0-.933.08-1.358.226.217-.696.299-1.311.242-1.85a.645.645 0 00-1.283.135c.075.708-.248 1.777-.997 3.171a.622.622 0 00-.057.148 4.148 4.148 0 00-.71 2.083l-.004.088v-.014a3.626 3.626 0 01-2.012-5.039 3.622 3.622 0 01-2.06-3.889 2.967 2.967 0 015.332-1.114c-.323.3-.567.704-.72 1.213a.645.645 0 101.235.373c.233-.774.742-1.043 1.769-.87a.645.645 0 10.215-1.272 3.995 3.995 0 00-1.126-.042l-.202.03-.088-.135a4.254 4.254 0 00-4.627-1.695 3.63 3.63 0 013.318-4.518h.015l.046-.186a3.624 3.624 0 013.495-2.667z"
      fill="#FFF"
    />
  </g>
</svg>
`;

exports[`<TechLogo /> renders a excel logo 1`] = `
<svg
  viewBox="0 0 39 39"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <circle
      cx="19.5"
      cy="19.5"
      fill="#05192D"
      r="19.5"
    />
    <path
      d="M20.971 9.016l-.001 1.986h8.28a.75.75 0 01.743.652l.007.102v14.502a.751.751 0 01-.75.753h-8.285v2.005L9 26.842V11.158l11.971-2.142zm-.001 4.06h2.832v1.817h-2.833v.693h2.833v1.817l-2.834-.001v.693h2.834v1.817h-2.835v.692h2.835v1.818l-2.836-.001v.697h2.836v1.734l-2.837-.001v1.468h8.343V11.691H20.97v1.385zm6.875 10.042v1.73H24.49v-1.73h3.355zM17.143 15.47l-1.476.068-1.061 2.506-.882-2.412-1.516.072 1.563 3.31-1.646 3.274 1.402.08 1.025-2.492 1.047 2.61 1.591.094-1.76-3.566 1.713-3.544zm10.702 5.135v1.817H24.49v-1.817h3.355zm0-2.51v1.817H24.49v-1.817h3.355zm0-2.509v1.817H24.49v-1.817h3.355zm0-2.51v1.817H24.49v-1.816h3.355z"
      fill="#FFF"
      fill-rule="nonzero"
    />
  </g>
</svg>
`;

exports[`<TechLogo /> renders a power_bi logo 1`] = `
<svg
  viewBox="0 0 39 39"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <circle
      cx="19.5"
      cy="19.5"
      fill="#05192D"
      r="19.5"
    />
    <path
      d="M14.874 24.518c.05.521.021 1.051.024 1.578.002.494.013.99-.005 1.484-.03.861-.588 1.437-1.354 1.42-.755-.017-1.236-.58-1.241-1.458-.003-.462 0-.925 0-1.388 0-.495-.006-.99.001-1.484.01-.758.506-1.329 1.19-1.381.692-.053 1.313.47 1.385 1.229zm2.648-5.973c.783-.007 1.315.618 1.317 1.556l.002 1.844v1.844c0 1.23.002 2.46-.002 3.689-.002.908-.525 1.515-1.299 1.521-.783.006-1.306-.586-1.307-1.503-.005-2.475-.004-4.95 0-7.424 0-.922.513-1.52 1.29-1.527zm3.95 2c.762.002 1.291.592 1.3 1.473.008.942.001 1.884.001 2.827 0 .894.007 1.788-.001 2.683-.009.88-.537 1.47-1.3 1.47-.755.002-1.296-.598-1.3-1.47a669.57 669.57 0 01.001-5.51c.004-.877.544-1.474 1.3-1.473zm3.93-5.266c.784-.004 1.3.585 1.302 1.509.004 1.787 0 3.574 0 5.361 0 1.82.006 3.639-.004 5.458-.004.68-.394 1.181-.983 1.325a1.24 1.24 0 01-1.452-.71 1.89 1.89 0 01-.148-.737c-.01-3.575-.01-7.15-.004-10.724.002-.88.538-1.478 1.29-1.482zm-8.627-4.276h11.288c1.792.002 2.936 1.208 2.936 3.091 0 3.195.002 6.39 0 9.585-.002 1.63-1.222 2.91-2.781 2.938-.312.006-.5-.064-.469-.447.025-.308-.077-.63.426-.643 1.155-.027 1.794-.737 1.796-1.942.005-3.195.006-6.39 0-9.585-.003-1.206-.705-1.938-1.859-1.938-5.743-.004-11.485-.005-17.228 0-1.142 0-1.835.72-1.838 1.9-.005 3.243-.006 6.486.001 9.729.003 1.045.652 1.797 1.611 1.835.647.025.606-.02.604.726 0 .093-.019.186-.028.273-1.151.348-2.585-.477-3.045-1.769a3.164 3.164 0 01-.18-1.025c-.013-3.274-.009-6.55-.007-9.824.001-1.633 1.179-2.89 2.741-2.9 2.01-.013 4.022-.004 6.032-.004z"
      fill="#FFF"
    />
  </g>
</svg>
`;

exports[`<TechLogo /> renders a python logo 1`] = `
<svg
  viewBox="0 0 39 39"
>
  <g
    fill="none"
    fill-rule="evenodd"
  >
    <circle
      cx="19.5"
      cy="19.5"
      fill="#05192D"
      r="19.5"
    />
    <path
      d="M26.849 14.552c1.432 0 2.107 1.055 2.46 2.457.492 1.948.514 3.408 0 4.926-.497 1.474-1.03 2.457-2.46 2.457h-7.376v.615h4.915v1.847c0 1.399-1.221 2.11-2.46 2.463-1.864.533-3.358.451-4.915 0-1.301-.377-2.46-1.15-2.46-2.463V22.24c0-1.328 1.113-2.463 2.46-2.463h4.915c1.638 0 3.074-1.404 3.074-3.073v-2.152zm-5.166 11.351a1.018 1.018 0 101.018 1.015c0-.561-.455-1.015-1.018-1.015zm-2.27-16.59a15.51 15.51 0 012.57.2c1.337.22 2.465 1.215 2.465 2.538v4.65a2.472 2.472 0 01-2.466 2.482h-4.925c-1.672 0-3.08 1.422-3.08 3.033v2.232H12.28c-1.433 0-2.27-1.03-2.62-2.476-.473-1.942-.453-3.103 0-4.963.392-1.623 1.647-2.476 3.08-2.476h6.782v-.62h-4.931V12.05c0-1.409.378-2.173 2.465-2.538a14.217 14.217 0 012.356-.2zm-2.678 1.747a1.02 1.02 0 00-1.019 1.021c0 .562.456 1.016 1.02 1.016a1.018 1.018 0 100-2.037z"
      fill="#FFF"
      fill-rule="nonzero"
    />
  </g>
</svg>
`;
